File: v1/src/main/java/com/api/v1/person/domain/PersonRepository.java
package com.api.v1.person.domain;

import org.springframework.data.mongodb.repository.Query;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

import java.util.UUID;

public interface PersonRepository extends ReactiveMongoRepository<Person, UUID> {

    @Query("{ 'ssn': ?0 }")
    Mono<Person> findBySsn(String ssn);

}

File: v2/src/main/kotlin/com/api/v2/persons/PersonRepository.kt
package com.api.v2.persons

import org.springframework.data.mongodb.repository.Query
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import java.util.UUID

interface PersonRepository: CoroutineCrudRepository<Person, UUID> {

    @Query("{ 'ssn': ?0 }")
    fun findBySsn(ssn: String): Person?

}